src/sample-executors/BaseExecutor.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ResolvedOrder, SignedOrder} from "../base/ReactorStructs.sol";
+import {IReactorCallback} from "../interfaces/IReactorCallback.sol";
+import {IReactor} from "../interfaces/IReactor.sol";
+import {Multicall} from "./Multicall.sol";
+
+struct PermitData {
+    address token;
+    bytes data;
+}
+
+abstract contract BaseExecutor is IReactorCallback, Multicall, Owned {
+    IReactor public immutable reactor;
+
+    constructor(IReactor _reactor, address _owner) Owned(_owner) {
+        reactor = _reactor;
+    }
+
+    /// @inheritdoc IReactorCallback
+    function reactorCallback(ResolvedOrder[] calldata, bytes calldata callbackData) external virtual;
+
+    function execute(SignedOrder memory order, bytes memory callbackData) public payable virtual {
```

Comment:
why is this virtual?

Reply:
made this virtual so we can add the `onlyWhitelistedCaller` modifier to it

Reply:
not a huge fan of this auth scheme, especially for `reactorCallback` because that **must** be restricted to only being called by the reactor but since its virtual in this base executor we can't enforce that here. So it's possible that integrators could make the mistake of not gating it by caller